hivelink-core/src/main/java/org/apache/iceberg/hivelink/core/LegacyHiveTable.java

```diff
@@ -118,7 +118,7 @@ public String location() {
 
   @Override
   public Snapshot currentSnapshot() {
-    throw new UnsupportedOperationException("Snapshots not supported for Hive tables without Iceberg metadata");
+    return null;
```

Comment:
Maybe for consistency purposes, also return `null` for `public Snapshot snapshot(long snapshotId)` and return an empty list for `public Iterable<Snapshot> snapshots()` ?

Reply:
IMO, returning null for `public Snapshot snapshot(long snapshotId)` would not be correct. User is requesting a specific snapshotId and so it is a clear user. WDYT?

I will change `snapshots()`

Reply:
Sure, I'm fine with this.

Reply:
Done